app/src/main/java/io/appium/uiautomator2/utils/DisplayIdHelper.java

```
@@ -0,0 +1,48 @@
+package io.appium.uiautomator2.utils;
+
+import android.annotation.SuppressLint;
+import android.view.Display;
+
+import java.lang.reflect.Method;
+
+import io.appium.uiautomator2.common.exceptions.UiAutomator2Exception;
+
+public class DisplayIdHelper {
+    /**
+     * Attempts to find the physical display ID that corresponds to the given Display.
+     *
+     * @param display android.view.Display instance
+     * @return matching physical display ID, or -1 if not found

```

These guard values might be confusing and error-prone. Either consider making the method Nullable or let it throw an exception

---

Addressed, please review.